tests/bool.rs

```diff
@@ -0,0 +1,26 @@
+#![cfg(feature = "std")]
+
+extern crate rand;
+extern crate libc;
+
+use rand::{FromEntropy, SmallRng};
+use rand::distributions::{Distribution, Bernoulli};
+
+#[link_name = "m"]
+extern "C" {
+    fn nextafter(x: f64, y: f64) -> f64;
+}
+
+/// This test should make sure that we don't accidentally have undefined
+/// behavior for large propabilties due to
+/// https://github.com/rust-lang/rust/issues/10184.
+#[test]
+fn large_probability() {
+    let p = unsafe { nextafter(1., 0.) };
```

Comment:
I think this should be `1 - ε/2` as [here](https://github.com/rust-lang-nursery/rand/blob/master/src/distributions/float.rs#L204). It would be nice not to link against an external lib unnecessarily (especially since it may not be available on all platforms).

Reply:
Can you explain why we would run into undefined behavior, as we stay well below `u64::MAX`?

Reply:
We would with the old implementation extended to 64 bits. `1.0*(u64::MAX as f64) as u64` is undefined behavior.